src/openlrr/game/front/Loader.cpp

```diff
@@ -22,18 +26,241 @@ LegoRR::Loader_Globs & LegoRR::loaderGlobs = *(LegoRR::Loader_Globs*)0x00556e00;
+
+	//Check if we're in the gamedata section and copy over
+	if ((secCntr == LOADER_MAXSECTIONS) && (localLoaderSec != nullptr)) {
+		uint32* allocPtr = (undefined4*)Gods98::Mem_Alloc(strlen(section));
+		localLoaderSec->name = (char*)allocPtr;
+		strcpy((char*)allocPtr, section);
+
+		localLoaderSec->currentSize = 0;
+		localLoaderSec->totalSize = 0;
+		loaderGlobs.current = localLoaderSec;
+	}
+
+	//Setup flags and create callback
+	loaderGlobs.flags = loaderGlobs.flags | LOADER_GLOB_FLAG_ENABLED;
+	loaderGlobs.progressLast = -1.0;
+	Loader_FileLoadCallback(nullptr, 0, nullptr);
+	return;
+}
 
 // <LegoRR.exe @0044e180>
-//void __cdecl LegoRR::Loader_FileLoadCallback(const char* filename, uint32 fileSize, void* data);
+void __cdecl LegoRR::Loader_FileLoadCallback(const char* filename, uint32 fileSize, void* data)
+{
+	int x,y;
+	long long lVar2;
+	Size2F barSize;
+	Point2F barPosition;
 
-// <LegoRR.exe @0044e360>
-//void __cdecl LegoRR::Loader_display_shutdown(void);
+	float localFloat = fileSize;
+
+	if (((byte)loaderGlobs.flags & LOADER_GLOB_FLAG_ENABLED) != 0) {
+		if (((loaderGlobs.current == nullptr) || ((loaderGlobs.current)->currentSize = (loaderGlobs.current)->currentSize + localFloat, loaderGlobs.current == nullptr))
+			|| ((loaderGlobs.current)->totalSize == 0)) {
+			localFloat = 0.0f;
+		}
+		else {
+			//Get percentage here
+			float num = loaderGlobs.current->currentSize;
+			float deom = loaderGlobs.current->totalSize;
+			localFloat = num/deom;
+		}
+		/* // minimum increase since last update of progress bar
+		   if (floatVal > (loaderGlobs.progressLast - 0.002f)) */
+		if (loaderGlobs.progressLast + 0.002f < localFloat) {
+			barPosition.y = loaderGlobs.ProgressWindow.y;
+			barPosition.x = loaderGlobs.ProgressWindow.x;
+			barSize.width = loaderGlobs.ProgressWindow.width;
+			barSize.height =loaderGlobs.ProgressWindow.height;
+			Gods98::Image_DisplayScaled(loaderGlobs.LoadScreen, nullptr, nullptr, nullptr);
+			/* if (floatVal > 1.0f) floatVal = 1.0f; */
+			if (1.0f < localFloat) {
+				/* 1.0f */
+				localFloat = 1.0f;
+			}
+			switch (loaderGlobs.ProgressDirection) {
+			case DIRECTION_UP:
+				barPosition.y = (barSize.height + barPosition.y) - barSize.height * localFloat;
+				barSize.height = (barSize.height + barPosition.y) - barPosition.y;
+				break;
+			case DIRECTION_RIGHT:
+				barSize.width = barSize.width * localFloat;
+				break;
+			case DIRECTION_DOWN:
+				barSize.height = barSize.height * localFloat;
+				break;
+			case DIRECTION_LEFT:
+				barPosition.x = (barSize.width + barPosition.x) - barSize.width * localFloat;
+				barSize.width = (barSize.width + barPosition.x) - barPosition.x;
+			}
+			if (loaderGlobs.ProgressBar !=  nullptr) {
+				Gods98::Image_DisplayScaled(loaderGlobs.ProgressBar, nullptr, &barPosition, (const Point2F*)&barSize);
+			}
+			if (loaderGlobs.LoadingText != nullptr) {
+				//center text in screen
+				x = (Gods98::appWidth()/2) - (loaderGlobs.LoadingWidth >> 1);
```

Comment:
Right shift by 1 (`loaderGlobs.LoadingWidth >> 1`) is just division by 2. Change it to `loaderGlobs.LoadingWidth / 2`.